### ISSUE TITLE
rasbian-buster-cross can not find python-apt

### ISSUE THREAD

**Issue by pakalvoltan** [lueschem/edi-raspbian#6]
Hello, I tried to use edi-raspbian but the set up fails when ansible asks the raspbian-buster-cross to update and upgrade apt. Here the message. Do you know if something else is needed for this to work? I follow the instructions and have tried several times:

After trying:
sudo edi -v lxc configure raspbian-buster-cross buster-cross.yml

at one point I get this error:

TASK [apt_setup : Update and upgrade apt.] *************************************
[WARNING]: Updating cache and auto-installing missing dependency: python-apt
fatal: [raspbian-buster-cross]: FAILED! => {"changed": false, "cmd": "apt-get install --no-install-recommends python-apt -y -q", "msg": "E: Package 'python-apt' has no installation candidate", "rc": 100, "stderr": "E: Package 'python-apt' has no installation candidate\n", "stderr_lines": ["E: Package 'python-apt' has no installation candidate"], "stdout": "Reading package lists...\nBuilding dependency tree...\nPackage python-apt is not available, but is referred to by another package.\nThis may mean that the package is missing, has been obsoleted, or\nis only available from another source\n\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Package python-apt is not available, but is referred to by another package.", "This may mean that the package is missing, has been obsoleted, or", "is only available from another source", ""]}

Thanks!

**Comment by lueschem** [comment]
"Update and upgrade apt." is where the setup tries to reach the internet from within the container for the first time.

Can you try to do the following:
```
lxc exec raspbian-buster-cross bash
apt update
apt install python-apt
```
I guess this will fail too but like this we get more information.
Are you working behind a proxy?

**Comment by pakalvoltan** [comment]
Thanks for your quick response Matthias!

I am not working behind a proxy. My Ubuntu 20.04 is directly connected to the network, but I see that the failure is that it can not resolve de.debian.org. 

I noticed that the container can not resolve the name:

`
 $ lxc exec raspbian-buster-cross bash
root@raspbian-buster-cross:~# apt update
Err:1 http://deb.debian.org/debian buster InRelease
  Temporary failure resolving 'deb.debian.org'
Reading package lists... Done    
Building dependency tree... Done
All packages are up to date.
W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease  Temporary failure resolving 'deb.debian.org'
W: Some index files failed to download. They have been ignored, or old ones used instead.
root@raspbian-buster-cross:~# ping deb.debian.org
ping: deb.debian.org: Temporary failure in name resolution
`

So I stopped my desktop firewall (I was using nftables), and now  apparently it can resolve the name, but it can not access it:

`
 # apt update
Err:1 http://deb.debian.org/debian buster InRelease          
  Could not connect to debian.map.fastlydns.net:80 (151.101.134.132), connection timed out Unable to connect to deb.debian.org:http:
Reading package lists... Done                                
Building dependency tree... Done
All packages are up to date.
W: Failed to fetch http://deb.debian.org/debian/dists/buster/InRelease  Could not connect to debian.map.fastlydns.net:80 (151.101.134.132), connection timed out Unable to connect to deb.debian.org:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.

`

I will continue debugging, but any light you can shed is greatly appreciated.

Thanks

**Comment by lueschem** [comment]
Strange - it looks like port 80 is still somehow blocked.

Maybe you could try this from within the container:

```
curl http://deb.debian.org/debian/dists/buster/InRelease 
```

and 

```
curl https://deb.debian.org/debian/dists/buster/InRelease 
```

**Comment by pakalvoltan** [comment]
Yes, I guess I have an lxc networking issue. I can ping 151.101.134.132 from the host, but not form within the container. 
I'll try to resolve it and post here when (if) I find the answer.

Thanks!

**Comment by pakalvoltan** [comment]
Hi Matthias, 
Your debugging tip helped me to fix the issue.  
Somehow my firewall messed up the 
`sudo /snap/bin/lxd init`

After stopping the firewall, I needed to clean all the lxc instances, bridges and profiles (delete all of them), and start from scratch with my firewall down.

After that I was able to pass the "apt update" task.  
I was able to get to this:

`Success: Configured container raspbian-buster-cross.
`

Thank you !
